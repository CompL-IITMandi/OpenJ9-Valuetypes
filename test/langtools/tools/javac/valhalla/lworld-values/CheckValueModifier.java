/*
 * @test /nodynamiccopyright/
 * @summary Compiler should reject value modifier when it features in illegal contexts.
 *
 * @compile/fail/ref=CheckValueModifier.out -XDrawDiagnostics CheckValueModifier.java
 */

/* Note: value as a modifier will be rejected by the parser if it features as a
   modifier of a (a) catch parameter, (b) resource variable, (c) for loop's init section
   declarators and (d) formal parameters. We test here only for the other illegal places.

   All uses of value below should trigger errors.
*/
class CheckValueModifier {
   inline int x;
   inline int foo() {
   }
   inline interface IFace {}
   inline @interface Annot {}
   inline enum Enum {}
   inline abstract class Inner {}
}
