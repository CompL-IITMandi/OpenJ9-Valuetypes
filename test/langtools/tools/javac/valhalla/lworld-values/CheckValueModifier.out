CheckValueModifier.java:18:11: compiler.err.illegal.combination.of.modifiers: interface, value
CheckValueModifier.java:19:12: compiler.err.illegal.combination.of.modifiers: interface, value
CheckValueModifier.java:20:11: compiler.err.mod.not.allowed.here: value
CheckValueModifier.java:21:20: compiler.err.illegal.combination.of.modifiers: abstract, value
CheckValueModifier.java:15:15: compiler.err.mod.not.allowed.here: value
CheckValueModifier.java:16:15: compiler.err.mod.not.allowed.here: value
6 errors
